ArjunaCore/arjuna/classes/com/arjuna/ats/arjuna/tools/osb/mbean/ObjStoreBrowser.java

```diff
@@ -80,14 +80,20 @@
     private static OSBTypeHandler[] defaultJTSOsbTypes = {
             new OSBTypeHandler(
                     true,
-                    false,
                     "com.arjuna.ats.internal.jta.tools.osb.mbean.jts.JTSXAResourceRecordWrapper",
                     "com.arjuna.ats.internal.jta.tools.osb.mbean.jts.JTSXAResourceRecordWrapper",
                     "CosTransactions/XAResourceRecord",
                     null
             ),
             new OSBTypeHandler(
                     true,
+                    "com.arjuna.ats.internal.jta.tools.osb.mbean.jts.JCAServerTransactionWrapper",
```

Comment:
Now both recovery (ServerTransactionRecoveryModule) and the tooling can potentially reactivate the same ServerTransaction. I am unclear about the consequences but for example they might both try to update the state resulting in a corrupted log. But this accusation could apply to any log record that is managed by the tooling. Strictly speaking any tooling attempt to update state should be coordinated with the recovery manager (either local or remote).

But you are correct that there is no way to remove the status otherwise (without a significant redesign).

But my main point was that we should be consistent about instrumenting this class and com.arjuna.ats.internal.jta.transaction.arjunacore.subordinate.jca.SubordinateAtomicAction. Currently, by default we provide a setting that the user must explicitly enable to access this instrumentation:

https://github.com/jbosstm/narayana/blob/master/ArjunaCore/arjuna/classes/com/arjuna/ats/arjuna/tools/osb/mbean/ObjStoreBrowserMBean.java#L22